modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/StructuredContentFolderResourceImpl.java

```diff
@@ -214,10 +241,42 @@ public StructuredContentFolder postSiteStructuredContentFolder(
 			parentStructuredContentFolderId);
 
 		return _addStructuredContentFolder(
+			structuredContentFolder.getExternalReferenceCode(),
 			journalFolder.getGroupId(), parentStructuredContentFolderId,
 			structuredContentFolder);
 	}
 
+	@Override
+	public StructuredContentFolder
+			putSiteStructuredContentFolderByExternalReferenceCode(
+				Long siteId, String externalReferenceCode,
+				StructuredContentFolder structuredContentFolder)
+		throws Exception {
+
+		JournalFolder journalFolder =
+			_journalFolderLocalService.
+				fetchJournalFolderByExternalReferenceCode(
+					siteId, externalReferenceCode);
+
+		if (journalFolder != null) {
+			return _toStructuredContentFolder(
+				_journalFolderService.updateFolder(
```

Comment:
Maybe we could refactor this method to a private method and invoke it from there and from the put by Id

Reply:
Makes sense to maintain the consistency 😁